superset/connectors/sqla/models.py

```diff
@@ -1086,7 +1086,7 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
 
         # To ensure correct handling of the ORDER BY labeling we need to reference the
         # metric instance if defined in the SELECT clause.
-        metrics_exprs_by_label = {m.name: m for m in metrics_exprs}
+        metrics_exprs_by_label = {m.key: m for m in metrics_exprs}
```

Comment:
refer to:
https://github.com/apache/superset/blob/b489cffb570c4d078ec982fd11c3e46dd0310f24/superset/connectors/sqla/models.py#L938-L943

Reply:
I'd like to have more documentation on this as comments in the code. Would you mind adding a comment for like 942 describing what the purpose is of `key`, and also here specifying why we use the `key` attribute?

Reply:
Thanks for the review! I have added comments here. `db_engine_spec` of Bigquery changed `metric_label`.

https://github.com/apache/superset/blob/0ed97eb6966ab49340459cb2c22b7c01bb43ff1d/superset/db_engine_specs/bigquery.py#L208